packages/aws-cdk/lib/cli/io-host/cli-io-host.ts

```diff
@@ -154,6 +168,45 @@ export class CliIoHost implements IIoHost {
     this.stackProgress = props.stackProgress ?? StackActivityProgress.BAR;
   }
 
+  /**
+   * Required for telemetry
+   */
+  public async bindTelemetryClient(argv: any, context: {[key: string]: any}) {
+    // TODO: change this to EndpointTelemetryClient
+    this.telemetryClient = new IoHostTelemetryClient({
+      ioHost: this,
+    });
+
+    // sanitize the raw cli input
+    const command = sanitizeCommandLineArguments(argv, await makeConfig());
+    this.telemetryInfo = {
+      identifiers: {
+        installationId: getInstallationId(this.asIoHelper()),
+        sessionId: randomUUID(),
+        telemetryVersion: '1.0',
+        cdkCliVersion: version.versionNumber(),
+        accountId: await new AccountIdFetcher().fetch(),
+        region: await new RegionFetcher().fetch(),
```

Comment:
cdklibraryversion

Reply:
put in todo -- requires cloud assembly OR package.json (and then we drop for non ts packages)

Reply:
Amplify achieves this by having an `NpmLockFileReader` class and a `YarnLockFileReader` class